compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -4232,7 +4232,7 @@ private void handleForeachDefinitionVariables(VariableDefinitionNode variableDef
         // Check whether the foreach node's variables are declared with var.
         if (isDeclaredWithVar) {
             // If the foreach node's variables are declared with var, type is `varType`.
-            handleDeclaredVarInForeach(variableNode, varType, blockEnv);
+            handleDeclaredVarInForeach(variableNode, Types.getReferredType(varType), blockEnv);
```

Comment:
Even if we do this; we'll lose the type ref information. My suggestion is to pass the reference type to `handleDeclaredVarInForeach()`. Extract out the actual type there; use it where needed. eg: type tag checks. Use the reference type in errors as in https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java#L1851 and https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java#L1890